File: v2/snippets/javascript/common/channelTests.js
const {Javonet} = require('javonet-nodejs-sdk')
const {describe, expect, test} = require("@jest/globals");
const path = require("path");
const ActivationCredentials = require("../../utils/ActivationCredentials")
const {TcpConnectionData, WsConnectionData} = require('javonet-nodejs-sdk')

describe('Nodejs channel tests', () => {

    const resourcesDirectory = path.resolve(__dirname, '../../../..') + '/testResources/configuration-file'

    test(`Test_Channel_InMemory_Success`, () => {
        // <InMemoryChannel>
        // use Activate only once in your app
        Javonet.activate(ActivationCredentials.yourLicenseKey)

        let communicationChannel = Javonet.inMemory()
        // use communicationChannel to create runtimes to interact with
        // </InMemoryChannel>
        expect(communicationChannel).not.toBeNull()
    })

    test(`Test_Channel_Tcp_Success`, () => {
        // <TcpChannel>
        // use Activate only once in your app
        Javonet.activate(ActivationCredentials.yourLicenseKey)

        let connectionData = new TcpConnectionData("127.0.0.1", 8080)
        let communicationChannel = Javonet.tcp(connectionData)
        // use communicationChannel to create runtimes to interact with
        // </TcpChannel>
        expect(communicationChannel).not.toBeNull()
    })

    test(`Test_Channel_WebSocket_Success`, () => {
        // <WebSocketChannel>
        // use Activate only once in your app
        Javonet.activate(ActivationCredentials.yourLicenseKey)

        let connectionData = new WsConnectionData("ws://127.0.0.1:8080/ws")
        let communicationChannel = Javonet.webSocket(connectionData)
        // use communicationChannel to create runtimes to interact with
        // </WebSocketChannel>
        expect(communicationChannel).not.toBeNull()
    })

    test(`Test_Channel_WithConfigurationFile_Success`, () => {
        // <WithConfigurationFile>
        // use Activate only once in your app
        Javonet.activate(ActivationCredentials.yourLicenseKey)

        // configuration to `withConfig` method can be provided in multiple ways:
        // - as path to JSON file (shown below),
        // - or directly as a JavaScript object,
        // - or as raw JSON string,
        
        // Set up variables
        let configFile = resourcesDirectory + "/channel-tests-config.json"
        let communicationChannel = Javonet.withConfig(configFile)
        
        // use communicationChannel to create runtimes to interact with
        // </WithConfigurationFile>
        expect(communicationChannel).not.toBeNull()
    })
})

File: v2/snippets/golang/common/channelTests.go
package common

import (
	"fmt"
	"os"
	"testing"

	"javonet.com/integrationTests/utils/activationcredentials"
	Javonet "javonet.com/javonet"
	"javonet.com/javonet/utils/connectiondata"
)

var resourcesDirectory string

func init() {
	cwd, _ := os.Getwd()
	resourcesDirectory = cwd + "/../../../../testResources/configuration-file"
}

func Test_Channel_InMemory_Success(t *testing.T) {
	// <InMemoryChannel>
	// use Activate only once in your app
	_, err := Javonet.Activate(activationcredentials.YourLicenseKey)
	if err != nil {
		fmt.Println("Error: " + err.Error())
	}

	communicationChannel := Javonet.InMemory()
	// use communicationChannel to create runtimes to interact with
	// </InMemoryChannel>
	if err != nil {
		t.Fatal("Error: " + err.Error())
	}
	if communicationChannel == nil {
		t.Fatal("communicationChannel is nil")
	}
}

func Test_Channel_Tcp_Success(t *testing.T) {
	// <TcpChannel>
	// use Activate only once in your app
	_, err := Javonet.Activate(activationcredentials.YourLicenseKey)
	if err != nil {
		fmt.Println("Error: " + err.Error())
	}

	connectionData := connectiondata.NewTcpConnectionData("127.0.0.1", 8080)
	communicationChannel := Javonet.Tcp(*connectionData)
	// use communicationChannel to create runtimes to interact with
	// </TcpChannel>
	if err != nil {
		t.Fatal("Error: " + err.Error())
	}
	if communicationChannel == nil {
		t.Fatal("communicationChannel is nil")
	}
}

func Test_Channel_WithConfigurationFile_Success(t *testing.T) {
	// <WithConfigurationFile>
	// use Activate only once in your app
	_, err := Javonet.Activate(activationcredentials.YourLicenseKey)
	if err != nil {
		fmt.Println("Error: " + err.Error())
	}

	// set up variables
	configFilePath := resourcesDirectory + "/channel-tests-config.json"
	communicationChannel := Javonet.WithConfig(configFilePath)
	// use communicationChannel to create runtimes to interact with
	// </WithConfigurationFile>
	if err != nil {
		t.Fatal("Error: " + err.Error())
	}
	if communicationChannel == nil {
		t.Fatal("communicationChannel is nil")
	}
}
